platforms/jvm/plugins-application/src/main/java/org/gradle/jvm/application/tasks/CreateStartScripts.java

```diff
@@ -397,4 +403,51 @@ private Iterable<String> getRelativePath(FileCollection path) {
         return path.getFiles().stream().map(input -> "lib/" + input.getName()).collect(Collectors.toCollection(Lists::newArrayList));
     }
 
+    /**
+     * Visible for internal use only!
+     *
+     * @return the application name property
+     * @since 9.0
+     */
+    @Internal
+    @Incubating
+    public Property<String> getApplicationNameProperty() {
+        return applicationName;
+    }
+
+    /**
+     * Visible for internal use only!
+     *
+     * @return the output dir
+     * @since 9.0
+     */
+    @Internal
+    @Incubating
+    public Property<File> getOutputDirProperty() {
+        return outputDir;
+    }
+
+    /**
+     * Visible for internal use only!
+     *
+     * @return the executable dir
+     * @since 9.0
+     */
+    @Internal
+    @Incubating
+    public Property<String> getExecutableDirProperty() {
+        return executableDir;
+    }
+
+    /**
+     * Visible for internal use only!
+     *
+     * @return the default JVM options
+     * @since 9.0
+     */
+    @Internal
+    @Incubating
+    public ListProperty<String> getDefaultJvmOptsProperty() {
+        return defaultJvmOpts;
+    }
```

Comment:
I didn't find a better way. The spring boot plugin depends on the convention mapping and we'll need to access the properties anyway if we want to provide a fix.